### DISCUSSION TITLE
Transforming const to the configured target environment

### DISCUSSION THREAD

**Discussion by pepicrft** [ElMassimo/vite_ruby#83] [Q&A] [answered]
[troubleshooting section]: https://vite-ruby.netlify.app/guide/troubleshooting.html

- [x] I have tried upgrading by running `bundle update vite_ruby`.
- [x] I have read the __[troubleshooting section]__ before opening an issue.

### Description 📖
When running `bin/vite build` on [this repository](https://github.com/craftweg/buildify), I get the following error that I can't trace back to the root of the issue:

```
Building with Vite ⚡️
 > vite.config.ts:12:0: error: Transforming const to the configured target environment ("es5") is not supported yet
    12 │ const version = "0.1.0";
       ╵ ~~~~~
   tsconfig.json:11:14: note: The target environment was set to "es5" here
    11 │     "target": "es5",
       ╵               ~~~~~

 > vite.config.ts:13:0: error: Transforming const to the configured target environment ("es5") is not supported yet
    13 │ const isDistEnv = process.env.RAILS_ENV === "production";
       ╵ ~~~~~
   tsconfig.json:11:14: note: The target environment was set to "es5" here
    11 │     "target": "es5",
       ╵               ~~~~~

 > vite.config.ts:15:0: error: Transforming const to the configured target environment ("es5") is not supported yet
    15 │ const bugsnagOptions = {
       ╵ ~~~~~
   tsconfig.json:11:14: note: The target environment was set to "es5" here
    11 │     "target": "es5",
       ╵               ~~~~~

failed to load config from /Users/pepibumur/src/github.com/craftweg/buildify/projects/app/vite.config.ts
error during build:
Error: Build failed with 3 errors:
vite.config.ts:12:0: error: Transforming const to the configured target environment ("es5") is not supported yet
vite.config.ts:13:0: error: Transforming const to the configured target environment ("es5") is not supported yet
vite.config.ts:15:0: error: Transforming const to the configured target environment ("es5") is not supported yet
```

For some reason `esbuild` is trying to parse the `tsconfig.json` and `vite.config.ts` file.

### Reproduction 🐞

1. Clone [this repository](https://github.com/craftweg/buildify)
2. `cd` into `projects/app`.
3. Install dependencies `bundle install && yarn install`.
4. Run `bin/vite build`

<details>
<summary>Vite Ruby Info</summary>
```
Running via Spring preloader in process 56570
bin/vite present?: true
vite_ruby: 1.2.11
vite_rails: 2.0.11
rails: 6.1.3.2
node: v16.0.0
npm: 7.10.0
yarn: 1.22.10
pnpm: 
ruby: ruby 3.0.1p64 (2021-04-05 revision 0fb782ee38) [x86_64-darwin20]

installed packages:
buildify-app@ /Users/pepibumur/src/github.com/craftweg/buildify/projects/app
├─┬ @vitejs/plugin-legacy@1.4.1
│ └── vite@2.3.5 deduped
├─┬ vite-plugin-bugsnag@1.0.2
│ └── vite@2.3.5 deduped
├─┬ vite-plugin-ruby@2.0.4
│ └── vite@2.3.5 deduped
└── vite@2.3.5
```
</details>

**Comment by ElMassimo** [comment, marked as answer]
[tsconfig.json]: https://esbuild.github.io/api/#tsconfig
[es5]: https://esbuild.github.io/content-types/#es5

Hi @pepibumur!

When asking for help please open a _Discussion_ instead—I intend to use _Issues_ for problems in the library only 😃 

<hr/>

Following the settings in [tsconfig.json] is a [feature of esbuild][tsconfig.json].

`es5` is now considered an old dialect and it's [not well supported in esbuild][es5], so it won't transpile the `es6` `const` to its `es5` equivalent.

Since all modern browsers support `es6` and more recent dialects, it's safe to target `es6` instead, which [you already did](https://github.com/craftweg/buildify/commit/9fb4688eaee10484006478fbe47827f274a7e9ee#diff-424f1263ebdc5b114f4eb9ebf96813535a5942d7b9d02c67507153cf84fbbc8cL11-R11), cool!

**Reply by pepicrft** [reply]
Thanks @ElMassimo and sorry for not posting it in the right place. Changing it to es6 made it pass.
